hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -2440,6 +2440,12 @@ public ClearDeadServersResponse clearDeadServers(RpcController controller,
         Set<Address> clearedServers = new HashSet<>();
         for (HBaseProtos.ServerName pbServer : request.getServerNameList()) {
           ServerName server = ProtobufUtil.toServerName(pbServer);
+          final boolean deadInProcess = master.getServerManager().isDeadServersInProgress(server);
+
+          if (!deadInProcess) {
+            throw new RuntimeException(String.format("Dead server '%s' is not 'dead' in fact...", server));
+          }
+
           if (!deadServer.removeDeadServer(server)) {
```

Comment:
You can filter the SCP here too? I thought no need to introduce new method. Please keep the code simple. Thanks.

Reply:
OK, Thanks for your suggestion